PurchasesCoreSwiftTests/Support/ManageSubscriptionsModalHelperTests.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ManageSubscriptionsModalHelperTests.swift
+//
+//  Created by Andrés Boedo on 8/20/21.
+
+import Foundation
+import Nimble
+import XCTest
+
+@testable import PurchasesCoreSwift
+
+#if os(macOS) || os(iOS)
+
+class ManageSubscriptionsModalHelperTests: XCTestCase {
+
+    private var systemInfo: MockSystemInfo!
+    private var purchaserInfoManager: MockPurchaserInfoManager!
+    private var identityManager: MockIdentityManager!
+    private var helper: ManageSubscriptionsModalHelper!
+    private let mockPurchaserInfoData: [String: Any] = [
+        "request_date": "2018-12-21T02:40:36Z",
+        "subscriber": [
+            "original_app_user_id": "app_user_id",
+            "first_seen": "2019-06-17T16:05:33Z",
+            "subscriptions": [:],
+            "other_purchases": [:],
+            "original_application_version": NSNull()
+        ],
+        "managementURL": NSNull()
+    ]
+
+
+    override func setUp() {
+        systemInfo = try! MockSystemInfo(platformFlavor: "", platformFlavorVersion: "", finishTransactions: true)
+        purchaserInfoManager = MockPurchaserInfoManager(operationDispatcher: MockOperationDispatcher(),
+                                                        deviceCache: MockDeviceCache(),
+                                                        backend: MockBackend(),
+                                                        systemInfo: systemInfo)
+        identityManager = MockIdentityManager(mockAppUserID: "appUserID")
+        helper = ManageSubscriptionsModalHelper(systemInfo: systemInfo,
+                                                purchaserInfoManager: purchaserInfoManager,
+                                                identityManager: identityManager)
+    }
+
+    func testShowManageSubscriptionModalMakesRightCalls() throws {
+        guard #available(iOS 15.0, *) else { return }
+        // given
+        var callbackCalled = false
+        purchaserInfoManager.stubbedPurchaserInfo = PurchaserInfo(data: mockPurchaserInfoData)
+        
+        // when
+        helper.showManageSubscriptionModal { result in
+            callbackCalled = true
+        }
+        
+        // then
+        expect(callbackCalled).toEventually(beTrue())
+        expect(self.purchaserInfoManager.invokedPurchaserInfo) == true
+        
+        // we'd ideally also patch the UIApplication (or NSApplication for mac), as well as
+        // AppStore, and check for the calls in those, but it gets very tricky.
```

Comment:
couldn't think of ways around this that weren't long, brittle and overly-complex, especially given that we use either UIApplication or NSWorkspace, and we use different things for different OS releases, all of them static, and with some like UIApplication being really tricky to mock (there can only be one instance of UIApplication in a running app, even tests)